derp/derphttp/derphttp_client.go

```diff
@@ -741,6 +745,17 @@ func (c *Client) dialNode(ctx context.Context, n *tailcfg.DERPNode) (net.Conn, e
 
 	nwait := 0
 	startDial := func(dstPrimary, proto string) {
+		dst := cmp.Or(dstPrimary, n.HostName)
+
+		// If dialing an IP address directly, check its address family
+		// and bail out before incrementing nwait.
+		if ip, err := netip.ParseAddr(dst); err == nil {
+			if proto == "tcp4" && ip.Is6() ||
+				proto == "tcp6" && ip.Is4() {
```

Comment:
I'm personally unfamiliar with this code, but is this intended to mismatch?

Reply:
Yes. It's saying "If we're doing an tcp4-specific dial, don't dial IPv6. And if we're doing a tcp6-specific dial, don't try IPv4." (we know both cases will just fail and cause an error)